tests/test_svg.py

```diff
@@ -0,0 +1,222 @@
+
+        # Make it more tricky
+        bsp_el2.elevate_degrees(0)
+        bsp_el2.elevate_degrees(1)
+        nur_el3.elevate_degrees(0)
+        nur_el3.elevate_degrees(1)
+        bsp_el2.insert_knots(1, [0.5])
+        nur_el3.insert_knots(1, [0.5])
+
+        # Set some show_options
+        nur_el3.show_options["control_points"] = False
+        bez_el0.show_options["c"] = "grey"
+        bez_el0.show_options["alpha"] = 0.5
+        bez_el0.show_options["control_mesh_lw"] = 0.2
+        bez_el0.show_options["control_mesh_c"] = "orange"
+        bez_el0.show_options["knot_c"] = "blue"
+        bsp_el2.show_options["control_point_ids"] = False
+        nur_el3.show_options["knots"] = False
+
+        # Test Output
+        with tempfile.TemporaryDirectory() as tmpd:
+            tmpf = c.to_tmpf(tmpd)
+
+            # With a list
+            c.splinepy.io.svg.export(
+                tmpf,
+                [bez_el0, rbz_el1, bsp_el2, nur_el3],
+                indent=False,
+                box_margins=0.7,
+            )
+
+            with open(tmpf) as tmp_read, open(
+                c.os.path.dirname(__file__) + "/data/svg_noindent.svg"
+            ) as base_file:
+                self.assertTrue(
+                    c.are_stripped_lines_same(
+                        base_file.readlines(), tmp_read.readlines(), True
+                    )
+                )
+
+            # Without a list
+            c.splinepy.io.svg.export(
+                tmpf,
+                bez_el0,
+                rbz_el1,
+                bsp_el2,
+                nur_el3,
+                indent=False,
+                box_margins=0.7,
+            )
+
+            with open(tmpf) as tmp_read, open(
+                c.os.path.dirname(__file__) + "/data/svg_noindent.svg"
+            ) as base_file:
+                self.assertTrue(
+                    c.are_stripped_lines_same(
+                        base_file.readlines(), tmp_read.readlines(), True
+                    )
+                )
+
+        # for python version > 3.9, test indented version
+        if int(python_version.split(".")[1]) >= 9:
+            with tempfile.TemporaryDirectory() as tmpd:
+                tmpf = c.to_tmpf(tmpd)
+                # With a list and also more options
+                c.splinepy.io.svg.export(
+                    tmpf,
+                    [bez_el0, rbz_el1, bsp_el2, nur_el3],
+                    indent=True,
+                    box_margins=0.5,
+                    # Set some show_options
+                    c="blue",
+                    alpha=0.2,
+                    control_point_r=0.1,
+                    knots=False,
+                    control_points=True,
+                    control_mesh_c="purple",
+                    control_point_c="black",
+                    control_mesh_lw=0.05,
+                )
+
+                with open(tmpf) as tmp_read, open(
+                    c.os.path.dirname(__file__) + "/data/svg_indent.svg"
```

Comment:
@clemens-fricke are you happy with this line?

Reply:
beautiful innit?

Reply:
I am blinded by the brilliance :D